CargoMonitor/CargoMonitor.cs

```diff
@@ -404,21 +409,22 @@ private void handleCommodityPurchasedEvent(CommodityPurchasedEvent @event)
 
         private bool _handleCommodityPurchasedEvent(CommodityPurchasedEvent @event)
         {
-            bool update = false;
-            Cargo cargo = GetCargoWithEDName(@event.commodityDefinition?.edname);
-            if (cargo != null)
+            Cargo cargo = GetCargoWithEDName(@event.commodityDefinition?.edname) ?? new Cargo(@event.commodityDefinition?.edname);
```

Comment:
Can we move 'new Cargo(...)' inside 'GetCargoWithEDName'?
I see the same pattern below and it could remove some duplication, but I do not know if this is used in different ways in other parts of the code.

Reply:
I count 6 references with methods which are conditional on whether the return value is null:
- _handleCommodityCollectedEvent
- _handleCommodityEjectedEvent
- _handleCommodityRefinedEvent
- handleCommoditySoldEvent
- _handleMissionCompletedEvent
- _handleEngineerContributedEvent

I also count 5 references where we instantiate `new Cargo` if the return is null. We could create a new method `GetOrCreateCargoWithEDName` to help distinguish this case if we think that doing so adds sufficient value.

Reply:
Well, in general any "get by name" lookup *should* return null if the item is not there, and it is the caller's responsibility to decide what to do about it. If we did make this change, we would definitely have to call it `GetOrCreateCargoWithEDName` instead, otherwise the function name is lying to us.

But on reflection I think I rather prefer having the error handling out there and in our faces in this case.